GVRf/Extensions/gvrf-particlesystem/src/main/java/org/gearvrf/particlesystem/GVREmitter.java

```diff
@@ -0,0 +1,349 @@
+    }
+
+    protected void tickClock(float time)
+    {
+        currTime = time;
+
+        for (GVRSceneObject partObj: mActiveParticleMeshes)
+        {
+            partObj.getRenderData().getMaterial().setFloat("u_time", time);
+        }
+    }
+
+    /**
+     * @param emitRate The rate( #particles/second ) at which this emitter emits particles.
+     *                 Currently clamped to 600 particles/second.
+     */
+    public void setEmitRate(int emitRate)
+    {
+        if ( emitRate > MAX_EMIT_RATE)
+            mEmitRate = MAX_EMIT_RATE;
+        else
+            mEmitRate = emitRate;
+    }
+
+    /**
+     * The rate ( #particles/second ) at which this emitter is emitting
+     * @return
+     */
+    public int getEmitRate()
+    {
+        return mEmitRate;
+    }
+
+    /**
+     * The ( emit rate * age ) decide the total number of particles in the scene.
+     * @param age The time (in seconds) the particles from this emitter will remain active in the
+     *            scene.
+     */
+    public void setParticleAge ( float age )
+    {
+        mMaxAge = age;
+    }
+
+    /**
+     * @param size the initial particle size.
+     */
+    public void setParticleSize ( float size )
+    {
+        mParticleSize = size;
+    }
+
+    /**
+     * The range of velocities that a particle generated from this emitter can have.
+     * @param minV Minimum velocity that a particle can have
+     * @param maxV Maximum velocity that a particle can have
+     */
+    public void setVelocityRange( float minV, float maxV )
+    {
+        minVelocity = minV;
+        maxVelocity = maxV;
+    }
+
+    /**
+     *
+     * @param acceleration the acceleration of the environment which affects the particle motion.
+     */
+    public void setEnvironmentAcceleration( Vector3f acceleration )
+    {
+        mEnvironmentAcceleration = acceleration;
+    }
+
+    /**
+     *
+     * @param rate The rate at which the particle size should increase or decrease. Lower value
+     *             clamped to 1.0f
+     */
+    public void setParticleSizeChangeRate( float rate )
+    {
+        mParticleSizeRate = rate;
+    }
+
+    /**
+     * @param fade Set this to true to linearly decrease the transparency of the particles with
+     *             their age.
+     */
+    public void setFadeWithAge ( boolean fade )
+    {
+        mFadeWithAge = fade;
+    }
+
+    /**
+     * @param mode True/false to enable/disable the burst mode for this emitter.
+     */
+
+    public void setBurstMode(boolean mode)
+    {
+        burstMode = mode;
+    }
+
+    /**
+     * @param tex Texture of the particle.
+     */
+    public void setParticleTexture(GVRTexture tex)
+    {
+        mParticleTexture = tex;
+    }
+
+    /**
+     *
+     * @param color The color value to be multiplied to the particle texture.
+     */
+    public void  setColorMultiplier( Vector4f color )
+    {
+        mColor = color;
+    }
+
+    public void clearSystem()
```

Comment:
Public methods need javadoc. Since this is the base class it really needs one. The derived would automatically inherit the javadoc supposedly. Also there is the {@inheritDoc} tag.